### DISCUSSION TITLE
Make Pages more optimized

### DISCUSSION THREAD

**Discussion by xzcollapse** [community/community#41767] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Please make the deployment process for github pages more optimized. I have a [repo](https://github.com/shadowgmes/shadowgmes.github.io) that has a HUGE filesize, and github pages takes about 10 minutes to deploy it. I tested it with Vercel, and it took THREE MINUTES. It cloned the repo to the server, then started a web server in that directory. Please make a process similar to that for pages deployments. I generally like github pages, and all github products for that matter, but please do something about this. Vercel is a bit slower than github, and I like github's experience with domains and management. Thanks, and sorry for the rant.

**Comment by yoannchaudet** [comment]
Thank you for posting here.

There are a few optimizations you can do yourself. First you are building your site as a Jekyll site which does not look like it is needed for you.

You could for instance switch your source to GitHub Actions and use our "static" template. More details in https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/.

Now if your site is so slow to deploy it is because it is too large. At over 2GB you are actually twice over our limit: https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#usage-limits.
